compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1430,23 +1431,34 @@ public BType transform(BLangFiniteTypeNode finiteTypeNode, AnalyzerData data) {
 
     @Override
     public BType transform(BLangTupleTypeNode tupleTypeNode, AnalyzerData data) {
-        List<BType> memberTypes = new ArrayList<>();
-        for (BLangType memTypeNode : tupleTypeNode.memberTypeNodes) {
-            BType type = resolveTypeNode(memTypeNode, data, data.env);
-
-            // If at least one member is undefined, return noType as the type.
-            if (type == symTable.noType) {
-                return symTable.noType;
-            }
-
-            memberTypes.add(type);
-        }
-
-        BTypeSymbol tupleTypeSymbol = Symbols.createTypeSymbol(SymTag.TUPLE_TYPE, Flags.asMask(EnumSet.of(Flag.PUBLIC)),
-                Names.EMPTY, data.env.enclPkg.symbol.pkgID, null,
+        List<BLangSimpleVariable> members = new ArrayList<>(tupleTypeNode.members.size());
+        BTypeSymbol tupleTypeSymbol = Symbols.createTypeSymbol(SymTag.TUPLE_TYPE,
+                Flags.asMask(EnumSet.of(Flag.PUBLIC)), Names.EMPTY, data.env.enclPkg.symbol.pkgID, null,
                 data.env.scope.owner, tupleTypeNode.pos, SOURCE);
-
-        BTupleType tupleType = new BTupleType(tupleTypeSymbol, memberTypes);
+        SymbolEnv tupleEnv = SymbolEnv.createTypeEnv(tupleTypeNode, new Scope(tupleTypeSymbol), data.env);
+        boolean hasUndefinedMember = false;
+        for (BLangSimpleVariable member : tupleTypeNode.members) {
+            BType bType = member.getBType();
+            if (bType == null) {
+                symbolEnter.defineNode(member, tupleEnv);
+            } else if (bType == symTable.noType) {
+                member.setBType(null);
```

Comment:
if `BType` is already set it type won't be resolved at SymbolEnter

Reply:
Can you check how it's handled for say records? Setting null doesn't look correct to me either.

Reply:
This is related to https://github.com/ballerina-platform/ballerina-lang/pull/39234#discussion_r1086230782 right? Since there is no flow in SymbolResolver for records to return `noType` if a field is undefined, it's fields can be defined later in https://github.com/ballerina-platform/ballerina-lang/blob/f715e82171cfac4a4d8010e27ff29177a66d532f/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java#L440 . How can we handle this in tuples?